### ISSUE TITLE
Feature request: Support for non-minibatch data for intrinsic functions

### ISSUE THREAD

**Issue by hholst80** [tensorflow/tensorflow#3611]
Some functions in the TensorFlow standard library makes an assumption that we're always feeding minibatches. 

One example is `tf.nn.softmax` which assumes a 2-D input of shape `[batch_size, num_classes]`. It would be nice to have a second API to such subroutines that work also without an assumption of a minibatch dimension in the input tensor.

**Comment by poxvoculi** [comment]
Could you give some more concrete examples of exactly what you'd like to do, and clarify whether it's merely awkward or impossible?

**Comment by hholst80** [comment]
Say `out` is a vector (one dimension tensor) and I would like to define a softmax output from it called `pi`:

The way I am doing it now:

```
    pi = tf.squeeze(tf.nn.softmax(tf.expand_dims(out, 0),
```

The way I would like to do it:

```
    pi = tf.nn.softmax(out)
```

**Comment by poxvoculi** [comment]
Thanks for clarifying.   This sounds like it could be useful, but is not urgent at the present time.  So it's a good candidate for a community contribution.
